nixos/modules/services/networking/prosody.nix

```diff
@@ -318,7 +324,8 @@ in
       serviceConfig = {
         User = "prosody";
         Type = "forking";
-        PIDFile = "/var/lib/prosody/prosody.pid";
+        RuntimeDirectory = [ "prosody" ];
```

Comment:
If I'm not mistaken, this will create a directory `/run/prosody`, but you're using `/var/run/prosody`. See https://www.freedesktop.org/software/systemd/man/systemd.exec.html.
And please add a comment what the option does so others don't have to look it up.